### PR TITLE
convert SourceBuild to use regular project files

### PR DESCRIPTION
**Pull request by brettfo**
Also add it to CI.

The result of running `eng/cibuild.sh /p:DotNetBuildFromSource=true /p:FSharpSourceBuild=true` is `artifacts/packages/$(Configuration)/Shipping/Microsoft.FSharp.Compiler.10.4.2.nupkg`.

### CODE REVIEW

**Comment by brettfo** [comment]
@chcosta Can you take a look at this?  After this, the next step is to handle dependency in-flow.

File: SourceBuild.sln [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿
```
Comment (chcosta) [inline SourceBuild.sln]: Did you explore the Arcade way of doing this which involves the [DotNetBuildFromSource](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L13) and [ExcludeFromSourceBuild](https://github.com/dotnet/arcade/blob/cdd5d318667c5af1740d69c8a7755e23e7964aeb/src/Microsoft.DotNet.Arcade.Sdk/sdk/Sdk.targets#L5) properties?

**Comment by chcosta** [comment]
FYI @crummel @dagood @tmat

File: SourceBuild.sln (hunk 1 above)
Comment (tmat) [reply SourceBuild.sln]: Yes, we have infrastructure for source build. No need to create a separate solution or build.cmd

Comment (brettfo) [reply SourceBuild.sln]: Thanks, working now on re-using our existing solutions and setting Exlcude where appropriate.

File: setup/Directory.Build.targets [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<Project>
+
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', '$(MSBuildThisFileDirectory)../'))" />
+
+  <Target Name="_IncludeProjectsForRestorePurposes"
```
Comment (brettfo) [inline setup/Directory.Build.targets]: Note: this doesn't fix the issue like I was hoping.  This file will be deleted.

**Pull request merged**